src/Streams/FileWriter.h

```diff
@@ -11,7 +11,16 @@ namespace Stream
 	class FileWriter : public SeekableWriter
 	{
 	public:
-		FileWriter(const std::string& filename);
+		// Open mode bit flags
+		enum OpenMode {
+			Default = 0,
+			FailIfExist = 0b0000'0001,
+			FailIfNoExist = 0b0000'0010,
+			PreserveContents = 0b0000'0100,
+			Append = 0b0000'1000,
+		};
```

Comment:
I'm a little confused by append and preserve contents. I expected both of these words to sort of mean the same thing in that the file would be opened and then seek to the end of the file.

-Brett

Reply:
The names need work. As for the meaning, basically don't truncate. You might want to update a file, without appending new data to it.

I suppose I could name it `Truncate`, and make it part of the default flag set (which is currently all bits clear).

Reply:
I think that truncate works well as the default behavior. It is typically what I want when saving to disk. Occasionally I might want to append, but that seems to be the not normal or default in work I've done.

Reply:
Yes, it was setup so the default would truncate.

It would be unusual to specify both truncation and appending, though I don't see that as necessarily wrong, just kind of odd. Sort of like specifying both needing to open an existing file, and needing to open a new file that doesn't previously exist. The result is quite predictable, maybe even useful in some rare cases (where the flags are variable parameters rather than hardcoded constants), but probably a bit silly most of the time.